File: infra/src/main/java/com/beat/infra/external/auth/social/kakao/client/KakaoApiClient.java
package com.beat.infra.external.auth.social.kakao.client;

import com.beat.infra.external.auth.social.kakao.response.KakaoUserResponse;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.http.HttpHeaders;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestHeader;

@FeignClient(name = "kakaoApiClient", url = "https://kapi.kakao.com")
public interface KakaoApiClient {

	@GetMapping(value = "/v2/user/me")
	KakaoUserResponse getUserInformation(@RequestHeader(HttpHeaders.AUTHORIZATION) String accessToken);
}

File: core/infra/src/main/kotlin/com/beat/infra/external/social/kakao/client/KakaoApiClient.kt
package com.beat.infra.external.social.kakao.client

import com.beat.infra.external.social.kakao.response.KakaoUserResponse
import org.springframework.cloud.openfeign.FeignClient
import org.springframework.http.HttpHeaders
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestHeader

@FeignClient(name = "kakaoApiClient", url = "https://kapi.kakao.com")
internal interface KakaoApiClient {
    @GetMapping(value = ["/v2/user/me"])
    fun getUserInformation(
        @RequestHeader(HttpHeaders.AUTHORIZATION) accessToken: String,
    ): KakaoUserResponse?
}
